File: educational-core/src/com/jetbrains/edu/learning/EduNames.java
/*
 * Copyright 2000-2014 JetBrains s.r.o.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.jetbrains.edu.learning;

import org.jetbrains.annotations.NonNls;

@NonNls
public class EduNames {

  public static final String PLUGIN_ID = "com.jetbrains.edu";

  public static final String JBA = "JetBrains Academy";

  public static final String EDU_PREFIX = "edu";
  public static final String TASK_HTML = "task.html";
  public static final String TASK_MD = "task.md";
  public static final String HINTS = "hints";
  public static final String LESSON = "lesson";
  public static final String FRAMEWORK_LESSON = "framework lesson";
  public static final String FRAMEWORK = "framework";
  public static final String SECTION = "section";
  public static final String TASK = "task";
  public static final String COURSE = "course";
  public static final String ITEM = "item";
  public static final String PROJECT = "project";
  public static final String TEST_TAB_NAME = "test";
  public static final String USER_TEST_INPUT = "input";
  public static final String USER_TEST_OUTPUT = "output";
  public static final String WINDOW_POSTFIX = "_window.";
  public static final String WINDOWS_POSTFIX = "_windows";
  public static final String ANSWERS_POSTFIX = "_answers";
  public static final String USER_TESTS = "userTests";
  public static final String TEST_HELPER = "test_helper.py";

  public static final String COURSE_META_FILE = "course.json";
  public static final String ADDITIONAL_MATERIALS = "Edu additional materials";
  public static final String PROJECT_NAME = "PROJECT_NAME";
  public static final String RUN_CONFIGURATION_DIR = "runConfigurations";

  public static final String DEFAULT_ENVIRONMENT = "";

  // Used as course type only
  public static final String PYCHARM = "PyCharm";
  public static final String EDU = "Edu";
  public static final String ANDROID = "Android";
  public static final String UNITTEST = "unittest";

  public static final String SRC = "src";
  public static final String TEST = "test";
  public static final String BUILD = "build";
  public static final String OUT = "out";

  public static final String STUDY_PROJECT_XML_PATH = "/.idea/study_project.xml";
  public static final String STEPIK_IDS_JSON = "stepik_ids.json";

  public static final String COURSE_IGNORE = ".courseignore";

  // IDs of supported languages. They are the same that `Language#getID` returns
  // but in some cases we don't have corresponding Language in classpath to get its id via `getID` method
  public static final String JAVA = "JAVA";
  public static final String KOTLIN = "kotlin";
  public static final String PYTHON = "Python";
  public static final String SCALA = "Scala";
  public static final String JAVASCRIPT = "JavaScript";
  public static final String TYPESCRIPT = "TypeScript";
  public static final String RUST = "Rust";
  // Single `ObjectiveC` id is used both for `ObjectiveC` and `C/C++`
  public static final String CPP = "ObjectiveC";
  public static final String GO = "go";
  public static final String PHP = "PHP";

  // Language versions
  public static final String PYTHON_2_VERSION = "2.x";
  public static final String PYTHON_3_VERSION = "3.x";

  // Submissions status
  public static final String CORRECT = "correct";
  public static final String WRONG = "wrong";
  public static final String UNCHECKED = "unchecked";

  // Troubleshooting guide links
  public static final String TROUBLESHOOTING_GUIDE_URL = "https://plugins.jetbrains.com/plugin/10081-edutools/docs/troubleshooting-guide.html";
  public static final String NO_TESTS_URL = TROUBLESHOOTING_GUIDE_URL + "#no_tests_have_run";
  public static final String NO_COURSES_URL = TROUBLESHOOTING_GUIDE_URL + "#no_courses_found";
  public static final String FAILED_TO_POST_TO_JBA_URL = TROUBLESHOOTING_GUIDE_URL + "#failed_submission_jba";
  public static final String FAILED_TO_CHECK_URL = TROUBLESHOOTING_GUIDE_URL + "#failed_to_launch_checking";
  public static final String OUTSIDE_OF_KNOWN_PORT_RANGE_URL = TROUBLESHOOTING_GUIDE_URL + "#outside_of_known_port_range";

  // Help links
  public static final String HELP_URL = "https://www.jetbrains.com/help";
  public static final String LEARNER_START_GUIDE = HELP_URL + "/education/learner-start-guide.html";

  private EduNames() {
  }

}

File: educational-core/src/com/jetbrains/edu/learning/EduNames.kt
package com.jetbrains.edu.learning

import com.jetbrains.edu.learning.courseFormat.EduFormatNames.TROUBLESHOOTING_GUIDE_URL

object EduNames {
  const val PLUGIN_ID = "com.jetbrains.edu"
  const val JBA = "Hyperskill"
  const val EDU_PREFIX = "edu"
  const val HINTS = "hints"

  const val FRAMEWORK_LESSON = "framework lesson"
  const val COURSE = "course"
  const val ITEM = "item"
  const val PROJECT = "project"
  const val WINDOW_POSTFIX = "_window."
  const val WINDOWS_POSTFIX = "_windows"
  const val ANSWERS_POSTFIX = "_answers"
  const val TEST_HELPER = "test_helper.py"
  const val COURSE_META_FILE = "course.json"
  const val ADDITIONAL_MATERIALS = "Edu additional materials"
  const val PROJECT_NAME = "PROJECT_NAME"
  const val RUN_CONFIGURATION_DIR = "runConfigurations"

  const val EDU = "Edu"
  const val ANDROID = "Android"
  const val UNITTEST = "unittest"
  const val SRC = "src"
  const val TEST = "test"
  const val BUILD = "build"
  const val OUT = "out"
  const val STUDY_PROJECT_XML_PATH = "/.idea/study_project.xml"
  const val STEPIK_IDS_JSON = "stepik_ids.json"
  const val COURSE_IGNORE = ".courseignore"
  const val CHANGE_NOTES = "change-notes.txt"

  // IDs of supported languages. They are the same that `Language#getID` returns
  // but in some cases we don't have corresponding Language in classpath to get its id via `getID` method
  const val JAVA = "JAVA"
  const val KOTLIN = "kotlin"
  const val PYTHON = "Python"
  const val SCALA = "Scala"
  const val JAVASCRIPT = "JavaScript"
  const val RUST = "Rust"

  // Single `ObjectiveC` id is used both for `ObjectiveC` and `C/C++`
  const val CPP = "ObjectiveC"
  const val GO = "go"
  const val PHP = "PHP"

  // Language versions
  const val PYTHON_2_VERSION = "2.x"
  const val PYTHON_3_VERSION = "3.x"

  // Troubleshooting guide links
  const val NO_COURSES_URL = "$TROUBLESHOOTING_GUIDE_URL#no_courses_found"
  const val CODEFORCES_ANTI_CRAWLER_URL = "$TROUBLESHOOTING_GUIDE_URL#codeforces_anti_crawler"
  const val FAILED_TO_POST_TO_JBA_URL = "$TROUBLESHOOTING_GUIDE_URL#failed_submission_jba"
  const val OUTSIDE_OF_KNOWN_PORT_RANGE_URL = "$TROUBLESHOOTING_GUIDE_URL#outside_of_known_port_range"

  const val ENVIRONMENT_CONFIGURATION_LINK_JAVA = "$TROUBLESHOOTING_GUIDE_URL#environment_configuration_java"
  const val ENVIRONMENT_CONFIGURATION_LINK_KOTLIN = "$TROUBLESHOOTING_GUIDE_URL#environment_configuration_kotlin"
  const val ENVIRONMENT_CONFIGURATION_LINK_PYTHON = "$TROUBLESHOOTING_GUIDE_URL#environment_configuration_python"
  const val ENVIRONMENT_CONFIGURATION_LINK_SCALA = "$TROUBLESHOOTING_GUIDE_URL#environment_configuration_scala"
  const val ENVIRONMENT_CONFIGURATION_LINK_JS = "$TROUBLESHOOTING_GUIDE_URL#environment_configuration_js"
  const val ENVIRONMENT_CONFIGURATION_LINK_RUST = "$TROUBLESHOOTING_GUIDE_URL#environment_configuration_rust"
  const val ENVIRONMENT_CONFIGURATION_LINK_CPP = "$TROUBLESHOOTING_GUIDE_URL#environment_configuration_cpp"
  const val ENVIRONMENT_CONFIGURATION_LINK_GO = "$TROUBLESHOOTING_GUIDE_URL#environment_configuration_go"
  const val ENVIRONMENT_CONFIGURATION_LINK_PHP = "$TROUBLESHOOTING_GUIDE_URL#environment_configuration_php"
  const val ENVIRONMENT_CONFIGURATION_LINK_ANDROID = "$TROUBLESHOOTING_GUIDE_URL#environment_configuration_android"
  const val ENVIRONMENT_CONFIGURATION_LINK_GRADLE = "$TROUBLESHOOTING_GUIDE_URL#environment_configuration_gradle"

  // Help links
  const val HELP_URL = "https://www.jetbrains.com/help"
  const val PLUGINS_HELP_LINK = "$HELP_URL/idea/managing-plugins.html"

  const val LEARNER_START_GUIDE = "$HELP_URL/education/learner-start-guide.html"
}